### PR TITLE
Adding roaring library.

### PR DESCRIPTION
**Pull request by lemire**
Specify library name and version:  **roaring/0.2.66**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

See http://roaringbitmap.org

### CODE REVIEW

File: recipes/roaring/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd and \
+           not self.settings.compiler.cppstd in self._supported_cppstd:
+          raise ConanInvalidConfiguration("This library requires c++11 standard or higher."
+                                          " {} required."
+                                          .format(self.settings.compiler.cppstd))
```
Comment (uilianries) [inline recipes/roaring/all/conanfile.py:50]:
```suggestion
        if self.settings.compiler.cppstd:
            check_min_cppstd(self, "11")
```

File: recipes/roaring/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    @property
+    def _build_subfolder(self):
+          return "build_subfolder"
+
+    @property
+    def _supported_cppstd(self):
+        return ["11", "gnu11", "14", "gnu14", "17", "gnu17", "20", "gnu20"]
+
+
```
Comment (uilianries) [inline recipes/roaring/all/conanfile.py:38]:
```suggestion
```

File: recipes/roaring/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import os
+import glob
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration, ConanException
```
Comment (uilianries) [inline recipes/roaring/all/conanfile.py:4]:
```suggestion
```

File: recipes/roaring/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import os
+import glob
```
Comment (uilianries) [inline recipes/roaring/all/conanfile.py:2]:
```suggestion
```

**Comment by lemire** [comment]
@uilianries  Change requests accepted.

**Comment by lemire** [comment]
@uilianries Thanks for the approval.

**Pull request merged**